content/articles/web-apps.md

```diff
@@ -0,0 +1,108 @@
+---
+author: "Emma Russell"
+date: 2023-09-12
+title: "RESIDE-IC Web Applications"
+tags:
+ - web
+---
+
+# Introduction
+
+This page describes some aspects to think about when considering a new RESIDE web application to present research:
+
+- why a research team might want a web application
+- whether a Shiny app or full web application would better meet their needs
+- the components of a full research web application
+- what we require from research in order to build an interface to it which a new web application can use
+- some different patterns of application we can provide
+
+# Why have a web application?
+..as opposed to making research code available for users to download and run themselves?
+
+Web apps are:
+- Accessible - no special skills required from user, independent of their coding knowledge
+- Independent of user's local system, available anywhere. Can deploy to appropriate compute resource.
+- Can deploy known stable versions - this helps with reproducible research
+- Usability - can design the user experience to focus on their concerns e.g. particular practical/policy goals, or a need to explore a research model in a guided way with appropriate visualisations.
+- Can provide a user model allowing users to easily save, manage and share their data
+- User does not need direct access to large datasets required for code execution
+- Can give researchers quick means to assess utility of a model before putting in effort to download, configure and run locally.
+
+
+## What about Shiny?
+Shiny is a popular choice for researchers wishing to provide web access to their models developed in R, but it has some drawbacks.
+
+Pros:
+- rapid web application development within R
+- easy to get started
+- powerful presentation features
+
+Cons:
+- difficult/impossible to automate tests
+- slow
+- does not scale easily
+- limited capabilities for many application features e.g. user model
```

Comment:
"user model" will be unfamiliar to researchers, but "password-based access", or "remembering what a user has done in previous sessions" might be?